.agents/skills/dsom-signature-injector/scripts/inject.py

```
@@ -0,0 +1,90 @@
+def inject_signature(target_path):
+    files_to_process = []
+
+    if os.path.isfile(target_path):
+        files_to_process.append(target_path)
+    elif os.path.isdir(target_path):
+        for root, _, files in os.walk(target_path):
+            if '.git' in root or os.path.join('.agents', 'brain') in root:
+                continue
+            for file in files:
+                if file.endswith(('.md', '.sh', '.ps1', '.yml', '.yaml')):
+                    files_to_process.append(os.path.join(root, file))
```

Ruff (0.16.0)

[warning] Use `list.extend` to create a transformed list

(PERF401)